### PR TITLE
rename detection service

### PR DESCRIPTION
**Pull request by xh3b4sd**
Towards Node Pools. We need detection services for clusters and node pools. This PR prepares that a little bit by renaming the current implementation which we use for clusters. This has still to be adapted once we get into the update magic.

### CODE REVIEW

File: service/controller/clusterapi/v29/cluster_resource_set.go [hunk 1]
```diff
@@ -91,13 +91,13 @@ func NewClusterResourceSet(config ClusterResourceSetConfig) (*controller.Resourc
 		}
 	}
 
-	var detectionService *detection.Detection
+	var clusterDetection *detection.Cluster
```
Comment (tuommaki) [inline service/controller/clusterapi/v29/cluster_resource_set.go]:
```suggestion
	var clusterChangeDetection *detection.ClusterChange
```

I'd like to have `Change` there as this is not about detecting clusters' presence as such but actual changes in them.

File: service/controller/clusterapi/v29/detection/cluster.go [hunk 2]
```diff
@@ -12,26 +12,26 @@ import (
...
 
-type Config struct {
+type ClusterConfig struct {
 	Logger micrologger.Logger
 }
 
-// Detection is a service implementation deciding if a tenant cluster should be
-// updated or scaled.
-type Detection struct {
+// Cluster is a detection service implementation deciding if a tenant cluster
```
Comment (tuommaki) [inline service/controller/clusterapi/v29/detection/cluster.go]:
```suggestion
// ClusterChange is a detection service implementation deciding if a tenant cluster
```

File: service/controller/clusterapi/v29/detection/cluster.go [hunk 3]
```diff
@@ -12,26 +12,26 @@ import (
...
+// should be updated or scaled.
```
Comment (tuommaki) [inline service/controller/clusterapi/v29/detection/cluster.go]:
```suggestion
// has changes in its desired state and if it should be updated or scaled thereafter.
```

File: service/controller/clusterapi/v29/cluster_resource_set.go (hunk 1 above)
Comment (xh3b4sd) [reply service/controller/clusterapi/v29/cluster_resource_set.go]: I would than rather like to rename the package to `changedetection`. This makes the package scope more specific and keeps the internals simpler.

Comment (tuommaki) [reply service/controller/clusterapi/v29/cluster_resource_set.go]: That makes sense :+1:

**Pull request merged**